### PR TITLE
Clean up docstrings in CMake files

### PR DESCRIPTION
**Pull request by jjwilke**
This gives more meaningful messages in the CMakeCache and ccmake. It also corrects a few inconsistencies between the documentation and the CMake files.

### CODE REVIEW

**Comment by jjwilke** [comment]
This is separate from Wiki updates and is mostly a CMake update.

File: cmake/kokkoskernels_tpls.cmake [hunk 1]
```diff
@@ -109,9 +119,9 @@ MACRO(kokkoskernels_find_header VAR_NAME HEADER TPL_NAME)
     SET(HAVE_CUSTOM_PATHS TRUE)
   ENDIF()
 
-  IF(NOT ${HEADER}_FOUND AND DEFINED KOKKOSKERNELS_${TPL_NAME}_DIR)
+  IF(NOT ${HEADER}_FOUND AND KOKKOSKERNELS_${TPL_NAME}_ROOT)
```
Comment (ndellingwood) [inline cmake/kokkoskernels_tpls.cmake]: What did _DIR need to be changed to _ROOT?

Comment (jjwilke) [reply cmake/kokkoskernels_tpls.cmake]: We could have left it? It seemed to better match "CMake style" to use _ROOT for the install prefix

**Comment by ndellingwood** [comment]
Spot-check results:

```bash
Running on machine: kokkos-dev-2

Repository Status:  7108e0915ff26df0e324f4eceb7c207b8f9460f3 ccmake docstring cleanup


Going to test compilers:  gcc/7.3.0 gcc/8.3.0 gcc/9.1 intel/18.0.5 clang/8.0 cuda/10.1
Testing compiler gcc/7.3.0
  Starting job gcc-7.3.0-OpenMP-release
kokkos options: 
kokkos devices: OpenMP
kokkos cxx: -O3 -Wall -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wignored-qualifiers -Wempty-body -Wclobbered -Wuninitialized 
  PASSED gcc-7.3.0-OpenMP-release
Testing compiler gcc/8.3.0
  Starting job gcc-7.3.0-Pthread-release
kokkos options: 
kokkos devices: Pthread
kokkos cxx: -O3 -Wall -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wignored-qualifiers -Wempty-body -Wclobbered -Wuninitialized 
  PASSED gcc-7.3.0-Pthread-release
Testing compiler gcc/9.1
  Starting job gcc-8.3.0-Serial-release
kokkos options: 
kokkos devices: Serial
kokkos cxx: -O3 -Wall -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wignored-qualifiers -Wempty-body -Wclobbered -Wuninitialized 
  PASSED gcc-8.3.0-Serial-release
  Starting job gcc-9.1-OpenMP-release
kokkos options: 
kokkos devices: OpenMP
kokkos cxx: -O3 -Wall -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wignored-qualifiers -Wempty-body -Wclobbered -Wuninitialized 
  PASSED gcc-9.1-OpenMP-release
Testing compiler intel/18.0.5
  Starting job gcc-9.1-Serial-release
kokkos options: 
kokkos devices: Serial
kokkos cxx: -O3 -Wall -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wignored-qualifiers -Wempty-body -Wclobbered -Wuninitialized 
  PASSED gcc-9.1-Serial-release
Testing compiler clang/8.0
  Starting job intel-18.0.5-OpenMP-release
kokkos options: 
kokkos devices: OpenMP
kokkos cxx: -O3 -Wall -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wuninitialized 
  PASSED intel-18.0.5-OpenMP-release
  Starting job clang-8.0-Cuda_OpenMP-release
kokkos options: 
kokkos devices: Cuda,OpenMP
kokkos cxx: -O3 -Wall -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wuninitialized -Wno-pass-failed 
  PASSED clang-8.0-Cuda_OpenMP-release
Testing compiler cuda/10.1
  Starting job clang-8.0-Pthread_Serial-release
kokkos options: 
kokkos devices: Pthread,Serial
kokkos cxx: -O3 -Wall -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wuninitialized -Wno-pass-failed 
  PASSED clang-8.0-Pthread_Serial-release
  Starting job cuda-10.1-Cuda_OpenMP-release
kokkos options: 
kokkos devices: Cuda,OpenMP
kokkos cxx: -O3 -Wall -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wuninitialized 
  PASSED cuda-10.1-Cuda_OpenMP-release
#######################################################
PASSED TESTS
#######################################################
clang-8.0-Cuda_OpenMP-release build_time=116 run_time=152
clang-8.0-Pthread_Serial-release build_time=49 run_time=154
cuda-10.1-Cuda_OpenMP-release build_time=150 run_time=140
gcc-7.3.0-OpenMP-release build_time=43 run_time=47
gcc-7.3.0-Pthread-release build_time=40 run_time=75
gcc-8.3.0-Serial-release build_time=42 run_time=70
gcc-9.1-OpenMP-release build_time=50 run_time=46
gcc-9.1-Serial-release build_time=44 run_time=74
intel-18.0.5-OpenMP-release build_time=123 run_time=46
```

File: cmake/kokkoskernels_tpls.cmake (hunk 1 above)
Comment (ndellingwood) [reply cmake/kokkoskernels_tpls.cmake]: Was just curious, thanks!

**Pull request merged**